### DISCUSSION TITLE
Lockfile maintenance for Terraform

### DISCUSSION THREAD

**Discussion by renan-alm** [renovatebot/renovate#10810] [Request Help] [answered]
Hi, 

I'm currently using a self-hosted instance of Renovate to monitor Terraform and I am willing to use the feature for management lockfiles (lockFileMaintenance or updateLockFiles).

Could you please assist on how to use the config env RENOVATE_X_TERRAFORM_LOCK_FILE along with either of the features mentioned above?

Thank you.
R.

**Comment by viceice** [comment, marked as answer]
simply  set `RENOVATE_X_TERRAFORM_LOCK_FILE=true` to renovate env to enable the lockfile feature

This only works for selh-hsted renovate

**Reply by stasostrovskyi** [reply]
Will it update lock file in the same PR(s) as provider updates? Or does it work the same as `lockFileMaintaince`, e.g running on a separate schedule?

**Reply by viceice** [reply]
i think it can only update lock file when providers change

**Reply by viceice** [reply]
also only repo root lockfile is currently supported

**Reply by stasostrovskyi** [reply]
Great to know, thank you! And last question - does it generate hashes for all supported platforms or only for the platform where renovate itself is running?

**Reply by viceice** [reply]
it should generate for all available platforms of a provider

**Reply by stasostrovskyi** [reply]
Cool, thank you very much for your help!

**Reply by secustor** [reply]
Lockfile updates are supported at provider changes as well as `lockFileMaintaince`. 
The lock file can be located anywhere in the repo as long it is in the same folder as the Terraform files containing the provider definitions. 
Currently only Hashicorp providers are supported and hashes for all available plattforms are set.   
Non-Hashicorp providers should be working in the near future. PR: https://github.com/renovatebot/renovate/pull/10619
